src/trace_processor/importers/etw/etw_parser.cc

```diff
@@ -241,19 +242,21 @@ void EtwParser::PushSchedSwitch(uint32_t cpu,
 StringId EtwParser::TaskStateToStringId(int64_t task_state_int) {
   const auto state = static_cast<uint8_t>(task_state_int);
   // Mapping for the different Etw states with their string description.
-  std::map<uint8_t, base::StringView> etw_states_map = {
-      {0x00, "Initialized"},     // INITIALIZED
-      {0x01, "R"},               // READY
-      {0x02, "Running"},         // RUNNING
-      {0x03, "Stand By"},        // STANDBY
-      {0x04, "T"},               // TERMINATED
-      {0x05, "Waiting"},         // WAITING
-      {0x06, "Transition"},      // TRANSITION
-      {0x07, "Deferred Ready"},  // DEFERRED_READY
-  };
-
-  return etw_states_map.find(state) != etw_states_map.end()
-             ? context_->storage->InternString(etw_states_map[state])
+  static base::NoDestructor<std::map<uint8_t, base::StringView>> etw_states_map(
```

Comment:
A map with <10 elements will be so much more efficient as an std::array + a linear scan. Please switch to that, we should avoid using NoDestructor as much as possible.